homeassistant/components/opnsense/__init__.py

```diff
@@ -36,42 +31,43 @@
 )
 
 
-def setup(hass, config):
-    """Set up the opnsense component."""
+@dataclass
+class OPNsenseData:
+    """Shared OPNsense data."""
 
-    conf = config[DOMAIN]
-    url = conf[CONF_URL]
-    api_key = conf[CONF_API_KEY]
-    api_secret = conf[CONF_API_SECRET]
-    verify_ssl = conf[CONF_VERIFY_SSL]
-    tracker_interfaces = conf[CONF_TRACKER_INTERFACE]
+    hass_config: dict
 
-    interfaces_client = diagnostics.InterfaceClient(
-        api_key, api_secret, url, verify_ssl
-    )
-    try:
-        interfaces_client.get_arp()
-    except APIException:
-        _LOGGER.exception("Failure while connecting to OPNsense API endpoint")
-        return False
 
-    if tracker_interfaces:
-        # Verify that specified tracker interfaces are valid
-        netinsight_client = diagnostics.NetworkInsightClient(
-            api_key, api_secret, url, verify_ssl
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the OPNsense component."""
+    if config.get(DOMAIN) is not None:
+        OPNsenseData.hass_config = config
```

Comment:
Please don't share this between config entries.  All shared data should be stored in `hass.data`